homeassistant/components/openweathermap/config_flow.py

```diff
@@ -0,0 +1,149 @@
+"""Config flow for OpenWeatherMap."""
+import logging
+
+from pyowm import OWM
+from pyowm.exceptions.api_call_error import APICallError
+from pyowm.exceptions.api_response_error import UnauthorizedError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_LANGUAGE,
+    DEFAULT_FORECAST_MODE,
+    DEFAULT_LANGUAGE,
+    DEFAULT_NAME,
+    FORECAST_MODES,
+    LANGUAGES,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OpenWeatherMapConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for OpenWeatherMap."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OpenWeatherMapOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            latitude = user_input[CONF_LATITUDE]
+            longitude = user_input[CONF_LONGITUDE]
+
+            await self.async_set_unique_id(f"{latitude}-{longitude}")
+            self._abort_if_unique_id_configured()
+
+            status = await _validate_owm_api(self.hass, user_input[CONF_API_KEY])
+
+            if not status.get("auth", True):
+                errors["base"] = "auth"
+            if not status.get("connection", True):
+                errors["base"] = "connection"
+
```

Comment:
It would be simpler to have `_validate_owm_api ` return `errors` or trap the exceptions directly here.  Right now the code is creating a data structure that only it consumes.